src/Database/QueryBuilder.php

```diff
@@ -181,4 +208,72 @@ public function count($columns = '*')
 
         return $result;
     }
+
+    /**
+     * Update a record in the database.
+     *
+     * @param  array $values
+     * @return int
+     */
+    public function update(array $values)
+    {
+        $this->clearMemoryCache($this->from);
+
+        return parent::update($values);
+    }
+
+    /**
+     * Delete a record from the database.
+     *
+     * @param  mixed $id
+     * @return int
+     */
+    public function delete($id = null)
+    {
+        $this->clearMemoryCache($this->from);
+
+        return parent::delete($id);
+    }
+
+    /**
+     * Insert a new record into the database.
+     *
+     * @param  array  $values
+     * @return bool
+     */
+    public function insert(array $values)
+    {
+        $this->clearMemoryCache($this->from);
+
+        return parent::insert($values);
+    }
+
+    /**
+     * Run a truncate statement on the table.
+     *
+     * @return void
+     */
+    public function truncate()
+    {
+        $this->clearMemoryCache($this->from);
+
+        parent::truncate();
+    }
+
+    /**
+     * Clear the memory cache.
+     *
+     * @param  string|null  $table
+     * @return \Illuminate\Database\Query\Builder|static
+     */
+    public function clearMemoryCache($table = null)
+    {
+        $cache = MemoryCache::instance();
+
+        if ($cache->enabled()) {
+            $table ? $cache->forget($table) : $cache->flush();
```

Comment:
I was thinking about use cases where you'd need to call this method. Clearing the current table's cache is imo never needed since the affecting statements always clear it after themselves. So I think there are only two use cases:

1. When you're using a join or a subquery on a table that's been recently updated:

```php
User::join('group', 'foo', '=', 'bar')->clearMemoryCache('group')->get();
```

2. When the query is so complex (multiple joins, subqueries etc...) that it makes more sense to delete the entire cache altogether.

But maybe having a separate method for that is really a better idea?